core/async_job/queue.ex

```diff
@@ -125,28 +118,70 @@ defmodule AntikytheraCore.AsyncJob.Queue do
+  end
+
+  defp attempt_job(%__MODULE__{jobs: jobs} = q,
+                   {{_, job_id} = job_key, index_running2} = _target_job_info,
+                   job,
+                   time,
+                   remaining,
+                   index_waiting) do
+    job2 = %AsyncJob{job | remaining_attempts: remaining - 1}
+    jobs2 = Map.put(jobs, job_id, {job2, time, :waiting})
+    index_waiting2 = :gb_sets.add(job_key, index_waiting)
+    %__MODULE__{q | jobs: jobs2, index_waiting: index_waiting2, index_running: index_running2}
+  end
+
   defunp move_now_runnable_jobs(q :: v[t], now_millis :: v[pos_integer]) :: t do
     %__MODULE__{q | brokers_to_notify: []}
     |> move_now_runnable_jobs_impl(now_millis)
   end
-  defp move_now_runnable_jobs_impl(%__MODULE__{jobs:              jobs,
-                                               index_waiting:     index_waiting,
-                                               index_runnable:    index_runnable,
-                                               brokers_waiting:   brokers_waiting,
-                                               brokers_to_notify: brokers_to_notify} = q,
+  defp move_now_runnable_jobs_impl(%__MODULE__{index_waiting: index_waiting} = q,
                                    now_millis) do
-    case take_smallest_with_earlier_timestamp(index_waiting, now_millis) do
-      nil                                     -> q
-      {{_, job_id} = job_key, index_waiting2} ->
-        jobs2           = Map.update!(jobs, job_id, fn {j, t, :waiting} -> {j, t, :runnable} end)
-        index_runnable2 = :gb_sets.add(job_key, index_runnable)
-        q2              = %__MODULE__{q | jobs: jobs2, index_waiting: index_waiting2, index_runnable: index_runnable2}
-        case brokers_waiting do
-          []       -> q2
-          [b | bs] -> %__MODULE__{q2 | brokers_waiting: bs, brokers_to_notify: [b | brokers_to_notify]}
-        end
-        |> move_now_runnable_jobs_impl(now_millis)
-    end
+    target_job_info = take_smallest_with_earlier_timestamp(index_waiting, now_millis)
+    move_now_runnable_job(q, target_job_info, now_millis)
+  end
+
+  defp move_now_runnable_job(q, nil = _target_job_info, _now_millis), do: q
+  defp move_now_runnable_job(%__MODULE__{jobs:            jobs,
+                                         index_runnable:  index_runnable,
+                                         brokers_waiting: brokers_waiting} = q,
+                                  {{_, job_id} = job_key, index_waiting2} = _target_job_info,
+                                  now_millis) do
+    jobs2 = Map.update!(jobs, job_id, fn {j, t, :waiting} -> {j, t, :runnable} end)
+    index_runnable2 = :gb_sets.add(job_key, index_runnable)
+    q2 = %__MODULE__{q | jobs: jobs2, index_waiting: index_waiting2, index_runnable: index_runnable2}
+    move_first_waiting_broker(q2, brokers_waiting)
+    |> move_now_runnable_jobs_impl(now_millis)
+  end
+
+  defp move_first_waiting_broker(q, [] = _brokers_waiting), do: q
```

Comment:
Same as https://github.com/access-company/antikythera/pull/51#discussion_r300550170.
```suggestion
  defp move_first_waiting_broker(q, []), do: q
```

Reply:
I fixed it https://github.com/access-company/antikythera/pull/51/commits/43b95be6693fa8736bf6caf8bd391781a4ccc141